packages/editor/src/store/selectors.js

```diff
@@ -1720,15 +1724,26 @@ export function __experimentalGetDefaultTemplateTypes( state ) {
  *
  * @return {Array} The template part areas.
  */
-export const __experimentalGetDefaultTemplatePartAreas = createSelector(
-	( state ) => {
-		const areas =
-			getEditorSettings( state )?.defaultTemplatePartAreas ?? [];
-		return areas.map( ( item ) => {
-			return { ...item, icon: getTemplatePartIcon( item.icon ) };
-		} );
-	},
-	( state ) => [ getEditorSettings( state )?.defaultTemplatePartAreas ]
+export const __experimentalGetDefaultTemplatePartAreas = createRegistrySelector(
+	( select ) =>
+		createSelector( () => {
+			deprecated(
+				"select('core/editor').__experimentalGetDefaultTemplatePartAreas",
+				{
+					since: '6.7',
+					alternative:
+						"select('core/core-data').getEntityRecord( 'root', '__unstableBase' )?.default_template_part_areas",
```

Comment:
Technically, missing the last piece of the logic:

```ts
areas.map( ( item ) => {
				return { ...item, icon: getTemplatePartIcon( item.icon ) };
			} );
```

Not sure if I should add this line under the alternative key or simply remove it altogether.